### DISCUSSION TITLE
How to set Indexing.Analyzers.Search.Default in c# code

### DISCUSSION THREAD

**Discussion by toochevere** [ravendb/ravendb#14996] [Q&A] [answered]
Hello,

In c# code I am trying to configure `Indexing.Analyzers.Search.Default` to be `WhitespaceAnalyzer` on a database so that auto indexes use the whitespace analyzer for full text search fields.   

I tried this: 
```
ravenDocumentStore.Maintenance.Send(new PutDatabaseSettingsOperation(
    dbName,
    new Dictionary<string, string> {
        { "Indexing.Analyzers.Search.Default", "WhitespaceAnalyzer" }
    }));
```

But it appears that the default analyzer is still being used.  I determined this by checking the results of a test query that forces the creation of an auto index.  My test query uses a common word ("and") and returns no results (as I would expect from the default analyzer if the word was being excluded for being too common).  A similar query using an index returns results when WhitespaceAnalyzer is used in the creation of the index. 

Is `PutDatabaseSettingsOperation` the correct way to set this type of setting?  If it is, is configuration key being passed incorrectly? 

Note that this is not a huge blocker for me, since I don't plan on using auto indexes.  But I would like to understand better how to do this correctly.

**Comment by arekpalinski** [comment, marked as answer]
Hi. Can it be that you didn't restart (enable / disable) a database after you applied that setting?

**Reply by arekpalinski** [reply]
Also note that you you need to reset an auto index you have already created because the analyzer is used when the actual indexing work takes place.

**Reply by toochevere** [reply]
Yep, that toggling on and off again was it.  🤦🏻‍♂️   In retrospect, it was clearly spelled out [in the sample given](https://ravendb.net/docs/article-page/5.4/csharp/client-api/operations/maintenance/configuration/database-settings-operation) in the docs.
